pkg/gds/certman/certs.go

```diff
@@ -751,9 +752,19 @@ vaspsLoop:
 		reissuanceDate := expirationDate.Add(-time.Hour * 240)
 		timeBeforeExpiration := time.Until(expirationDate)
 
-		// TODO: handle the case where the certificate has expired, we should update the certificate record to the EXPIRED state
 		// NOTE: This computation returns fractional days rather than rounding up or down to the nearest day.
 		switch daysBeforeExpiration := timeBeforeExpiration.Hours() / 24; {
+
+		// If a certificate has expired, update the certificate record.
+		case daysBeforeExpiration <= 0:
+			var cert *models.Certificate
+			if cert, err = c.db.RetrieveCert(RetrieveCertID(identityCert)); err != nil {
+				log.Error().Err(err).Str("vasp_id", vasp.Id).Msg("could not update expired certificate record status")
+				continue vaspsLoop
+			}
+			cert.Status = models.CertificateState_EXPIRED
+			c.db.UpdateCert(cert)
```

Comment:
We need to handle any errors returned by `UpdateCert` here, my first instinct is that we don't need to bail out of the loop or anything but I think we should at least log the error. What do you think?

Reply:
Fixed! I hadn't notice UpdateCert returning an error, my bad!